tests/Resources/ti.geolocation.test.js

```diff
@@ -0,0 +1,289 @@
+		should(Ti.Geolocation.activityType).be.eql(Ti.Geolocation.ACTIVITYTYPE_FITNESS);
+	});
+
+	it.ios('.allowsBackgroundLocationUpdates', function () {
+		should(Ti.Geolocation).have.a.property('allowsBackgroundLocationUpdates').which.is.a.Boolean;
+		should(Ti.Geolocation.allowsBackgroundLocationUpdates).be.eql(false); // defaults to false (unless a special tiapp property is set, see docs)
+	});
+
+	it.ios('#getAllowsBackgroundLocationUpdates()', function () {
+		should(Ti.Geolocation).have.a.property('getAllowsBackgroundLocationUpdates').which.is.a.Function;
+		should(Ti.Geolocation.getAllowsBackgroundLocationUpdates()).be.a.Boolean;
+	});
+
+	it.ios('#setAllowsBackgroundLocationUpdates()', function () {
+		should(Ti.Geolocation).have.a.property('setAllowsBackgroundLocationUpdates').which.is.a.Function;
+		Ti.Geolocation.setAllowsBackgroundLocationUpdates(true); // defaults to false, set to true
+		should(Ti.Geolocation.allowsBackgroundLocationUpdates).be.eql(true);
+	});
+
+	// Intentionally skip for Android, doesn't exist
+	it.androidMissing('.distanceFilter', function () {
+		should(Ti.Geolocation).have.a.property('distanceFilter').which.is.a.Number;
+	});
+
+	it.androidMissing('#getDistanceFilter()', function () {
+		should(Ti.Geolocation).have.a.property('getDistanceFilter').which.is.a.Function;
+		should(Ti.Geolocation.getDistanceFilter()).be.a.Number;
+	});
+
+	it.androidMissing('#setDistanceFilter()', function () {
+		should(Ti.Geolocation).have.a.property('setDistanceFilter').which.is.a.Function;
+		Ti.Geolocation.setDistanceFilter(1000);
+		should(Ti.Geolocation.distanceFilter).eql(1000);
+	});
+
+	it.ios('.hasCompass', function () {
+		should(Ti.Geolocation).have.a.property('hasCompass').which.is.a.Boolean;
+	});
+
+	it.ios('#getHasCompass()', function () {
+		should(Ti.Geolocation).have.a.property('getHasCompass').which.is.a.Function;
+		should(Ti.Geolocation.getHasCompass()).be.a.Boolean;
+	});
+
+	// Intentionally skip for Android, doesn't exist
+	it.androidMissing('.headingFilter', function () {
+		should(Ti.Geolocation).have.a.property('headingFilter').which.is.a.Number;
+	});
+
+	it.androidMissing('#getHeadingFilter()', function () {
+		should(Ti.Geolocation).have.a.property('getHeadingFilter').which.is.a.Function;
+		should(Ti.Geolocation.getHeadingFilter()).be.a.Number;
+	});
+
+	it.androidMissing('#setHeadingFilter', function () {
+		should(Ti.Geolocation).have.a.property('setHeadingFilter').which.is.a.Function;
+		Ti.Geolocation.setHeadingFilter(90);
+		should(Ti.Geolocation.headingFilter).eql(90);
+	});
+
+	// https://jira.appcelerator.org/browse/TIMOB-26452
+	it.iosBroken('.lastGeolocation', function () {
+		should(Ti.Geolocation).have.a.property('lastGeolocation'); // TODO: which is a String/null/undefined?
+	});
+
+	it('#getLastGeolocation()', function () {
+		should(Ti.Geolocation).have.a.property('getLastGeolocation').which.is.a.Function;
+		const returnValue = Ti.Geolocation.getLastGeolocation();
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-default;
  File: tests/Resources/ti.geolocation.test.js;
  Line: 138;
-->

- :warning: tests/Resources/ti.geolocation.test.js line 138 – 'returnValue' is assigned a value but never used. (no-unused-vars)